kernel/src/loader/mod.rs

```diff
@@ -33,6 +34,34 @@ pub enum Error {
     SeekElfStart,
     SeekProgramHeader,
 }
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            match *self {
+                Error::BigEndianElfOnLittle => "Cannot convert Big Endian ELF to Little Endian",
+                Error::CommandLineCopy => "Failed to copy the command line string to guest memory",
+                Error::CommandLineOverflow => "Command line string overflows guest memory",
+                Error::InvalidElfMagicNumber => "Invalid ELF magic number",
+                Error::InvalidEntryAddress => "Invalid entry address found in ELF header",
+                Error::InvalidProgramHeaderSize => "Invalid ELF program header size",
+                Error::InvalidProgramHeaderOffset => "Invalid ELF program header offset",
+                Error::InvalidProgramHeaderAddress => "Invalid ELF program header address",
+                Error::ReadElfHeader => "Failed to read ELF header from the input VM image",
```

Comment:
I think we want to say since this error will only be thrown when trying to load the kernel image:
```suggestion
                Error::ReadElfHeader => "Failed to read ELF header",
```